cpp/include/cudf/hashing/detail/murmurhash3_x64_128.cuh

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <cudf/detail/utilities/hash_functions.cuh>
+
+#include <thrust/pair.h>
+
+namespace cudf::hashing::detail {
+
+// MurmurHash3_x64_128 implementation from
+// https://github.com/aappleby/smhasher/blob/master/src/MurmurHash3.cpp
+//-----------------------------------------------------------------------------
+// MurmurHash3_64_128 was written by Austin Appleby, and is placed in the public
```

Comment:
The upstream just calls it MurmurHash3. Let’s replicate this change for the other MurmurHash3 algorithm too. https://github.com/aappleby/smhasher/blob/61a0530f28277f2e850bfc39600ce61d02b518de/src/MurmurHash3.cpp#L2
```suggestion
// MurmurHash3 was written by Austin Appleby, and is placed in the public
```

Reply:
I disagree. That file contains all the MurmurHash algorithms and we need to distinguish them somehow, right?
https://github.com/aappleby/smhasher/blob/61a0530f28277f2e850bfc39600ce61d02b518de/src/MurmurHash3.cpp#L94
https://github.com/aappleby/smhasher/blob/61a0530f28277f2e850bfc39600ce61d02b518de/src/MurmurHash3.cpp#L150
https://github.com/aappleby/smhasher/blob/61a0530f28277f2e850bfc39600ce61d02b518de/src/MurmurHash3.cpp#L255

Reply:
I was thinking about it differently. I wanted to avoid editing the text of the license that we're copying: https://github.com/aappleby/smhasher/blob/61a0530f28277f2e850bfc39600ce61d02b518de/src/MurmurHash3.cpp#L2

The codebase as a whole is called `MurmurHash3`, and that's how it's licensed.